# wyrm/de-morgan

ğŸ“ Enforce using De Morgan's law to simplify negated logical expressions.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

Negated conjunction:

```tsx
!(foo && 24);

// Automatically fixed to:
!foo || !24;
```

Negated disjunction:

```tsx
!(foo || 24);

// Automatically fixed to:
!foo && !24;
```

### Correct âœ…

Logical expression with negated components:

```tsx
!foo || !24;
```

<!-- end auto-generated rule header -->
