# wyrm/distribute-boolean-casts

ðŸ“ Enforce that boolean casts are distributed over logical expressions.

ðŸ’¼ This rule is enabled in the following configs: ðŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ðŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

An entire logical expression wrapped in a boolean cast:

```tsx
const foo = !!(bar && baz.length > 2 && quux.description);

// Automatically fixed to:
const foo = !!bar && baz.length > 2 && !!quux.description;
```

### Correct âœ…

Distributed boolean cast:

```tsx
const foo = !!bar && baz.length > 2 && !!quux.description;
```

<!-- end auto-generated rule header -->
