# wyrm/duplicate-destructuring

ğŸ“ Forbid duplicate keys in object destructuring patterns.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Duplicate identifier key:

```tsx
const { length: bar, length: baz } = [];
```

### Correct âœ…

No duplicate key:

```tsx
const { map: at, at: map } = [];
```

<!-- end auto-generated rule header -->
