# wyrm/exact-string-regex

ğŸ“ Forbid using a RegEx when string equality would suffice.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ’¡ This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

## Cases

### Incorrect âŒ

Exact string RegEx:

```tsx
/^quux$/.test(foo);

// Can be fixed to:
foo === 'quux';
```

### Correct âœ…

Using string equality:

```tsx
foo === 'quux';
```

<!-- end auto-generated rule header -->
