# wyrm/idiomatic-cast

ğŸ“ Enforce idiomatic ways to cast values.

ğŸ’¼ This rule is enabled in the â˜‘ï¸ `strictTypeChecked` config.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

ğŸ’­ This rule requires [type information](https://typescript-eslint.io/linting/typed-linting).

## Cases

### Incorrect âŒ

Using `String()` with a number:

```tsx
declare const foo: number;
String(foo);

// Automatically fixed to:
declare const foo: number;
foo.toString();
```

### Correct âœ…

Using `.toString()` with a number:

```tsx
declare const foo: number;
foo.toString();
```

Using `String()` with a nullable number:

```tsx
declare const foo: number | undefined;
String(foo);
```

<!-- end auto-generated rule header -->
