# wyrm/no-commented-out-comment

ğŸ“ Forbid commented out comments.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Commented out comment:

```tsx
// // foo
```

Commented out block comment:

```tsx
// /* foo */
```

### Correct âœ…

Normal comment:

```tsx
// foo
```

Normal block comment:

```tsx
/* foo */
```

<!-- end auto-generated rule header -->
