# Forbid duplicated branches with early returns (`wyrm/no-duplicated-return`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Return value is the same for the early return and the final return:

```tsx
function foo() {
  if (Math.random()) return null;
  return null;
}
```

### Correct âœ…

No duplicated return:

```tsx
function foo() {
  if (Math.random()) return null;
  if (Math.random()) return null;
  return 42;
}
```

<!-- end auto-generated rule header -->
