# Forbid unnecessary `else` block after a `continue` statement (`wyrm/no-else-continue`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

Unnecessary `else` block after `continue`:

```tsx
while (true) {
  if (cond) continue;
  else foo();
}
```

### Correct âœ…

No `else` block necessary:

```tsx
while (true) {
  if (cond) continue;
  foo();
}
```

<!-- end auto-generated rule header -->
