# wyrm/no-empty-literal-iteration

ðŸ“ Forbid iterating over empty literals.

ðŸ’¼ This rule is enabled in the following configs: ðŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ðŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

`for..of` loop over an empty array:

```tsx
for (const x of []) {
}
```

`for..of` loop over an empty string:

```tsx
for (const x of '') {
}
```

`for..in` loop over an empty object:

```tsx
for (const x in {}) {
}
```

`.forEach()` over an empty array:

```tsx
[].forEach(() => {});
```

`.map()` over an empty array:

```tsx
[].map(() => {});
```

`.reduce()` over an empty array:

```tsx
[].reduce(() => {});
```

`.some()` over an empty array:

```tsx
[].some(() => {});
```

### Correct âœ…

`for..of` loop over a non-empty array literal:

```tsx
for (const x of [1, 2, 3]) {
}
```

<!-- end auto-generated rule header -->

## Related

- [`sonarjs/no-empty-collection`](https://sonarsource.github.io/rspec/#/rspec/S4158/javascript)
