# Forbid unnecessary `else` block after a `throw` statement (`wyrm/no-else-throw`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

Unnecessary `else` block after `throw`:

```tsx
if (cond) throw Error('oh no!');
else foo();
```

### Correct âœ…

No `else` block necessary:

```tsx
if (cond) throw Error('oh no!');
foo();
```

<!-- end auto-generated rule header -->

## Related

- [wyrm/no-else-break](https://github.com/mchevestrier/eslint-plugin-wyrm/blob/master/packages/eslint-plugin-wyrm/docs/rules/no-else-break.md)
- [wyrm/no-else-continue](https://github.com/mchevestrier/eslint-plugin-wyrm/blob/master/packages/eslint-plugin-wyrm/docs/rules/no-else-continue.md)
- ESLint: [no-else-return](https://eslint.org/docs/latest/rules/no-else-return)

## Prior Art

- Pylint: [no-else-raise](https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/no-else-raise.html)
