# Forbid unnecessary `else` block after a `throw` statement (`wyrm/no-else-throw`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

Unnecessary `else` block after `throw`:

```tsx
if (cond) throw Error('oh no!');
else foo();
```

### Correct âœ…

No `else` block necessary:

```tsx
if (cond) throw Error('oh no!');
foo();
```

<!-- end auto-generated rule header -->
