# wyrm/no-empty-jsx-expression

ğŸ“ Forbid empty JSX expression containers.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Empty JSX expression container:

```tsx
function Foo() {
  return (
    <div>
      {}
      Ok
    </div>
  );
}
```

JSX expression container with only a literal `null`:

```tsx
function Foo() {
  return <div>{null}</div>;
}
```

JSX expression container with only a literal `undefined`:

```tsx
function Foo() {
  return <div>{undefined}</div>;
}
```

JSX expression container with only a literal `false`:

```tsx
function Foo() {
  return <div>{false}</div>;
}
```

JSX expression container with only a literal empty string:

```tsx
function Foo() {
  return <div>{''}</div>;
}
```

### Correct âœ…

JSX expression container is not empty:

```tsx
function Foo({ children }: PropsWithChildren) {
  return <div>{children}</div>;
}
```

JSX expression container with a comment:

```tsx
function Foo() {
  return (
    <div>
      {/* A comment */}
      Ok
    </div>
  );
}
```

<!-- end auto-generated rule header -->
