# wyrm/no-float-length-check

ðŸ“ Forbid comparing a length to a floating point number.

ðŸ’¼ This rule is enabled in the following configs: ðŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ðŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Comparing a length to a floating point number:

```tsx
foo.length > 3.14;
```

Comparing a size to a floating point number:

```tsx
foo.size === 3.14;
```

### Correct âœ…

Integer length comparison:

```tsx
foo.length > 3;
```

Integer size comparison:

```tsx
foo.size >= 42;
```

<!-- end auto-generated rule header -->
