# wyrm/no-huge-useeffect

ðŸ“ Forbid huge `useEffect` functions.

ðŸ’¼ This rule is enabled in the following configs: ðŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

`useEffect` block spanning 21 lines:

```tsx
useEffect(() => {
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
}, []);
```

### Correct âœ…

`useEffect` block spanning 20 lines:

```tsx
useEffect(() => {
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
  foo();
}, []);
```

<!-- end auto-generated rule header -->

## Options

<!-- begin auto-generated rule options list -->

| Name         | Description                                                       | Type   |
| :----------- | :---------------------------------------------------------------- | :----- |
| `maxNbLines` | Maximum number of lines for a `useEffect` function. Default: `20` | Number |

<!-- end auto-generated rule options list -->
