# wyrm/no-if-length-for

ðŸ“ Forbid redundant condition for positive length before a loop.

ðŸ’¼ This rule is enabled in the following configs: ðŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ðŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ðŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

`for..of` loop with array length truthiness check:

```tsx
if (arr.length) {
  for (const x of arr) {
  }
}

// Automatically fixed to:
for (const x of arr) {
}
```

With `> 0`:

```tsx
if (arr.length > 0) {
  for (const x of arr) {
  }
}

// Automatically fixed to:
for (const x of arr) {
}
```

With `!== 0`:

```tsx
if (arr.length !== 0) {
  for (const x of arr) {
  }
}

// Automatically fixed to:
for (const x of arr) {
}
```

### Correct âœ…

`for..of` loop with no condition:

```tsx
for (const x of arr) {
}
```

<!-- end auto-generated rule header -->
