# wyrm/no-invalid-date-literal

ğŸ“ Disallow invalid date literals.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Invalid date:

```tsx
new Date('20-07-1969');
```

With a number:

```tsx
new Date(9007199254740991);
```

With `Date.parse()`:

```tsx
Date.parse('not a valid date');
```

With `Temporal.PlainDate`:

```tsx
Temporal.PlainDate.from('1969-20-07');
```

### Correct âœ…

Valid date:

```tsx
new Date('07-20-1969');
```

Valid date (with `Date.parse()`):

```tsx
Date.parse('07-20-1969');
```

<!-- end auto-generated rule header -->
