# Forbid JSX expression statements (`wyrm/no-jsx-statement`)

ğŸ’¼ This rule is enabled in the following configs: âœ… `wyrm/recommended`, â˜‘ï¸ `wyrm/recommendedTypeChecked`, ğŸŸ¢ `wyrm/strict`, ğŸŸ£ `wyrm/strictTypeChecked`.

## Cases

### Incorrect âŒ

```tsx
// JSX fragment in expression statement

export function MyComponent() {
  <></>;
}
```

### Correct âœ…

```tsx
// JSX element in return statement

export function MyComponent() {
  return <div />;
}
```

```tsx
// JSX element in variable initialization

export function MyComponent() {
  const jsx = <div />;
  return jsx;
}
```

<!-- end auto-generated rule header -->
