# wyrm/no-jsx-statement

ğŸ“ Forbid JSX expression statements.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

JSX element in expression statement:

```tsx
export function MyComponent() {
  <div />;
}
```

JSX fragment in expression statement:

```tsx
export function MyComponent() {
  <></>;
}
```

### Correct âœ…

JSX element in return statement:

```tsx
export function MyComponent() {
  return <div />;
}
```

JSX element in variable initialization:

```tsx
export function MyComponent() {
  const jsx = <div />;
  return jsx;
}
```

<!-- end auto-generated rule header -->
