# wyrm/no-numbered-comments

ğŸ“ Forbid numbered comments.

## Cases

### Incorrect âŒ

Inline numbered comment:

```tsx
// 23. Do stuff
```

Block numbered comment:

```tsx
/* 23. Do stuff */
```

### Correct âœ…

Not a numbered comment:

```tsx
// Ok
```

JSDoc numbered comment:

```tsx
/** 23. Do stuff */
```

<!-- end auto-generated rule header -->
