# Forbid redundant function declarations (`wyrm/no-redundant-function-declaration`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

Redundant function declaration with the same name:

```tsx
const foo = function foo() {};

// Automatically fixed to:
function foo() {}
```

Redundant function declaration with different names:

```tsx
const foo = function bar() {};

// Automatically fixed to:
function foo() {}
```

### Correct âœ…

Normal function declaration:

```tsx
function foo() {}
```

<!-- end auto-generated rule header -->
