# wyrm/no-return-to-void

ğŸ“ Forbid returning values in void-returning callbacks.

ğŸ’¼ This rule is enabled in the â˜‘ï¸ `strictTypeChecked` config.

ğŸ’­ This rule requires [type information](https://typescript-eslint.io/linting/typed-linting).

## Cases

### Incorrect âŒ

With return from `forEach` callback:

```tsx
[1, 2, 3].forEach((it) => {
  return 42;
});
```

With `return undefined` from `forEach` callback:

```tsx
[1, 2, 3].forEach((it) => {
  return undefined;
});
```

With some undefined return from `forEach` callback:

```tsx
[1, 2, 3].forEach((it) => {
  if (Math.cos(0)) return 42;
  return undefined;
});
```

### Correct âœ…

With no return from `forEach` callback:

```tsx
[1, 2, 3].forEach((it) => {
  console.log(it);
});
```

With empty return from `forEach` callback:

```tsx
[1, 2, 3].forEach((it) => {
  return;
});
```

<!-- end auto-generated rule header -->
