# Forbid suspicious semicolons in JSX (`wyrm/no-suspicious-jsx-semicolon`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Description

This rule detects the potential accidental insertion of a semicolon or comma in JSX when refactoring.

## Cases

### Incorrect âŒ

A suspicious-looking semicolon:

```tsx
export function MyComponent() {
  return (
    <div>
      <div>With a trailing semicolon at the end of the line</div>;
    </div>
  );
}
```

A suspicious-looking comma:

```tsx
export function MyComponent() {
  return (
    <div>
      <div>With a trailing comma at the end of the line</div>,
    </div>
  );
}
```

### Correct âœ…

No suspicious semicolon or comma:

```tsx
export function MyComponent() {
  return <div />;
}
```

<!-- end auto-generated rule header -->
