# wyrm/no-self-object-assign

ğŸ“ Forbid using `Object.assign()` with the same object as both target and source.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

`Object.assign()` to self:

```tsx
const obj = {};
Object.assign(obj, obj);
```

### Correct âœ…

With different identifiers:

```tsx
const obj1 = {};
const obj2 = {};
Object.assign(obj1, obj2);
```

<!-- end auto-generated rule header -->
