# wyrm/no-unbound-catch-error

ğŸ“ Forbid `catch` clauses with unbound errors.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Unbound error in `catch` clause:

```tsx
try {
  // ...
} catch {
  // ...
}
```

### Correct âœ…

Bound error in `catch` clause:

```tsx
try {
  // ...
} catch (err) {
  // ...
}
```

<!-- end auto-generated rule header -->

## Related

- ESLint: [`preserve-caught-error`](https://eslint.org/docs/latest/rules/preserve-caught-error#requirecatchparameter)
