# wyrm/no-useless-ts-check

ğŸ“ Forbid useless `@ts-check` comments in TypeScript files.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Useless `@ts-check` comment:

```tsx
// File name: foo.ts

// @ts-check
```

### Correct âœ…

Useful `@ts-check` comment in a JS file:

```tsx
// File name: foo.js

// @ts-check
```

<!-- end auto-generated rule header -->
