# Forbid useless IIFEs (`wyrm/no-useless-iife`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ’¡ This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

## Cases

### Incorrect âŒ

Useless IIFE with arrow function:

```tsx
const x = (() => 2)();
```

Useless IIFE (wrapped imperative logic):

```tsx
function quux() {
  (() => {
    foo();
    bar();
    baz();
  })();
}
```

### Correct âœ…

Useful IIFE:

```tsx
const foo = (() => {
  if (bar) return 42;
  if (baz) return 17;
  return 105;
})();
```

<!-- end auto-generated rule header -->
