# Forbid unassigned TODO comments (`wyrm/no-unassigned-todo`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Unassigned TODO comment:

```tsx
// TODO
```

### Correct âœ…

Assigned TODO comment:

```tsx
// TODO(foo)
```

<!-- end auto-generated rule header -->
