# Forbid leading or trailing whitespace in object keys (`wyrm/no-whitespace-property`)

ğŸ’¼ This rule is enabled in the following configs: ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Leading whitespace:

```tsx
const obj = { ' foo': 42 };
```

Trailing whitespace:

```tsx
const obj = { 'foo ': 42 };
```

TypeScript object type declaration:

```tsx
type Obj = { ' foo': 42 };
```

### Correct âœ…

Whitespace inside text:

```tsx
const obj = { 'foo bar': 42 };
```

<!-- end auto-generated rule header -->
