# wyrm/nullish-object-spread

ğŸ“ Forbid useless fallback for nullish values in object spread.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Useless fallback with empty object:

```tsx
const foo = null;
const obj = { ...(foo ?? {}) };
```

### Correct âœ…

No useless fallback:

```tsx
const foo = null;
const obj = { ...foo };
```

<!-- end auto-generated rule header -->
