# wyrm/optional-call-expression

ğŸ“ Enforce using optional call expression syntax.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

`if` statement:

```tsx
if (foo) {
  foo();
}

// Automatically fixed to:
foo?.();
```

### Correct âœ…

Optional call expression:

```tsx
foo?.();
```

<!-- end auto-generated rule header -->
