# wyrm/prefer-has

ğŸ“ Enforce using `Map#has` and `Set#has`.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Cases

### Incorrect âŒ

With `!== undefined`:

```tsx
foo.get('bar') !== undefined;

// Automatically fixed to:
foo.has('bar');
```

With `typeof !== 'undefined'`:

```tsx
typeof foo.get('bar') !== 'undefined';

// Automatically fixed to:
foo.has('bar');
```

### Correct âœ…

With `.has()`:

```tsx
foo.has('bar');
```

With `.get()` not being check for undefined:

```tsx
foo.get('bar');
```

<!-- end auto-generated rule header -->
