# wyrm/suspicious-map-length

ðŸ“ Disallow suspicious use of `.map().length`.

ðŸ’¼ This rule is enabled in the following configs: ðŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ðŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ðŸ’¡ This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

## Cases

### Incorrect âŒ

Suspicious use of `.map().length`:

```tsx
export const z = [1, 2].map((x) => x > 2).length;

// Can be fixed to:
export const z = [1, 2].filter((x) => x > 2).length;
```

### Correct âœ…

Correct use of `.filter().length`:

```tsx
export const z = [1, 2].filter((x) => x > 2).length;
```

<!-- end auto-generated rule header -->
