# wyrm/unused-object-assign

ğŸ“ Disallow unused `Object.assign()` expressions.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸ© `recommended`, âœ… `recommendedTypeChecked`, ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

## Cases

### Incorrect âŒ

Unused `Object.assign()`:

```tsx
Object.assign({}, { foo: 42 });
```

### Correct âœ…

Used `Object.assign()`:

```tsx
const fnord = Object.assign({}, { foo: 42 });
```

`Object.assign()` with side-effect:

```tsx
const quux = { bar: 105 };
Object.assign(quux, { foo: 42 });
```

<!-- end auto-generated rule header -->
