# wyrm/useless-intermediary-variable

ğŸ“ Disallow unnecessary intermediary variables.

ğŸ’¼ This rule is enabled in the following configs: ğŸŸª `strict`, â˜‘ï¸ `strictTypeChecked`.

ğŸ’¡ This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

## Cases

### Incorrect âŒ

Declaring another variable initialized with a simple identifier:

```tsx
const foo = 42;
const bar = foo;
return bar + 3;

// Can be fixed to:
const bar = 42;

return bar + 3;
```

### Correct âœ…

No useless intermediary variable:

```tsx
const bar = 42;
return bar + 3;
```

<!-- end auto-generated rule header -->
